### ISSUE TITLE
omicron builds fail when clippy is unhappy with code from dependencies

### ISSUE THREAD

**Issue by Nieuwejaar** [oxidecomputer/omicron#4888]
When trying to bump the `dendrite` version, `omicron` CI builds started failing because `clippy` was unhappy with some of the code imported from `dpd-client`:

```
2098    2024-01-23T21:44:13.181Z            --> /work/oxidecomputer/omicron/target/debug/build/dpd-client e0a00b57212e7804/out/dpd-client.rs:3333:9
2099    2024-01-23T21:44:13.181Z             |
2100    2024-01-23T21:44:13.181Z        3333 | /         let mut query = Vec::with_capacity(1usize);
2101    2024-01-23T21:44:13.181Z        3334 | |         query.push(("force_sync", force_sync.to_string()));
2102    2024-01-23T21:44:13.181Z             | |___________________________________________________________^ help: consider using the `vec![]` macro: `let query = vec![..];`
2103    2024-01-23T21:44:13.181Z             |
2104    2024-01-23T21:44:13.181Z             = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#vec_init_then_push
2105    2024-01-23T21:44:13.182Z             = note: `-D clippy::vec-init-then-push` implied by `-D warnings`
2106    2024-01-23T21:44:13.182Z             = help: to override `-D warnings` add `#[allow(clippy::vec_init_then_push)]`
```
(from the build log at https://github.com/oxidecomputer/omicron/runs/20791167193)

Ideally `progenitor` would generate clippy-clean code (https://github.com/oxidecomputer/progenitor/issues/701) or include a directive to clippy to ignore it.  Regardless, it seems odd that we would be running `clippy` on external crates as part of the `omicron` build process.

**Comment by ahl** [comment]
> Ideally `progenitor` would generate clippy-clean code ([oxidecomputer/progenitor#701](https://github.com/oxidecomputer/progenitor/issues/701)) or include a directive to clippy to ignore it. Regardless, it seems odd that we would be running `clippy` on external crates as part of the `omicron` build process.

Agreed that that would be ideal; I don't, however, think the benefit warrants the effort. Clippy has several lints that make sense for hand crafted code, but create many extra cases in generated code for little benefit. It could, of course, emit attributes to ignore known violated lints. We've done this by hand elsewhere: https://github.com/oxidecomputer/oxide.rs/blob/main/sdk/src/lib.rs#L11
